src/interfaces/token.ts

```diff
@@ -60,25 +60,31 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
    * @param address
    */
   async balances(address: Address): Promise<Balance[]> {
-    const vaultBalances = await this.yearn.vaults
-      .balances(address)
-      .then(balances => balances.filter(token => token.balance !== "0"));
-
-    if (this.chainId === 1 || this.chainId === 1337) {
-      let zapperBalances = await this.yearn.services.zapper.balances(address);
-      const vaultBalanceAddresses = new Set(vaultBalances.map(balance => balance.address));
-      zapperBalances = zapperBalances.filter(balance => !vaultBalanceAddresses.has(balance.address));
-      return zapperBalances.concat(vaultBalances);
-    }
+    try {
+      const vaultBalances = await this.yearn.vaults
+        .balances(address)
+        .then(balances => balances.filter(token => token.balance !== "0"));
+
+      if (this.chainId === 1 || this.chainId === 1337) {
+        let zapperBalances = await this.yearn.services.zapper.balances(address);
```

Comment:
I think the calls to zapper are the only ones we need to handle more gracefully - with this try/catch enclosing over all of this, if the call to zapper fails then we won't return the tokens from the vaults i.e. the line below, so if zapper fails then they'll still be able to deposit into vaults using the underlying token.

Reply:
My domain knowledge is still quite weak :) I thought we didn't want to do anything in case the zapper service would fail. Fixed with https://github.com/yearn/yearn-sdk/pull/241/commits/65cac31109b24fcda34373ac856cdf332874f44b